### ISSUE TITLE
[Bug] Cannot login after upgrade due to "incorrect integer value" of secure_only

### ISSUE THREAD

**Issue by teledyn** [piwik/piwik#22004]
### What happened?

Matomo was upgraded to 5.0.0 by a novice admin, and since then they have been unable to login. Logs show the INSERT INTO values do not match the table datatypes.

Checking in the mysql, that field in =mt67_user_token_auth= is =tiny-int=, so the /code/ should post 0 or =NULL=, and perhaps not =false=?

### What should happen?

The login =INSERT INTO= should match the database definition and succeed, and we should be able to login.

### How can this be reproduced?

It is the login screen.

### Matomo version

5.0.0

### PHP version

PHP 8.1.27

### Server operating system

Ubuntu 20.04.6 LTS

### What browsers are you seeing the problem on?

Firefox, Chrome, Other

### Computer operating system

android, windows, linux, osx

### Relevant log output

```shell
[05-Mar-2024 16:37:25 UTC] [modusresearch.com] Error in Matomo (tracker): Error query: Mysqli statement execute error : Incorrect integer value: '' for column 'secure_only' at row 1                                    In query: INSERT INTO mt67_user_token_auth (login, description, password, date_created, date_expired, system_token, hash_algo, secure_only) VALUES (?, ?, ?, ?, ?, ?, ?, ?)                                    Parameters: array (   0 => 'admin',   1 => 'System generated CliMultiNonAsyncArchive',   2 => '(redacted)',   3 => '2024-03-05 16:37:25',   4 => '2024-03-07 04:37:25',   5 => 1,   6 => 'sha512',   7 => false, )
```

### Validations

- [X] Read our [Contributing Guidelines](https://github.com/matomo-org/matomo/blob/5.x-dev/CONTRIBUTING.md).
- [X] Follow our [Security Policy](https://github.com/matomo-org/matomo/blob/5.x-dev/SECURITY.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating duplicates.
- [X] The provided steps to reproduce is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the Bug.

**Comment by teledyn** [comment]
If it wasn't clear in my first report, since the upgrade, we are unable to login. If there are any further tests or diagnostics you need, please let me know. Do I need to patch your code? Should I modify the datatype in the table?

**Comment by Starker3** [comment]
Hi @teledyn 
Could you try following the steps:
1. Follow the steps in this FAQ to make sure all files are updated correctly to Matomo 5: https://matomo.org/faq/on-premise/update-matomo/#the-manual-three-step-update
2. Execute this SQL query on the database directly:
`UPDATE matomo_option set option_value = '4.14.2' where option_value like '5.%';`
3. Then execute the `./console core:update` command manually on the server

The above steps would force Matomo to re-run all update commands on the database.

(Some additional info here: https://matomo.org/faq/how-do-i-downgrade-from-matomo-5-to-matomo-4/)

**Comment by teledyn** [comment]
your advice actually turned out to be very useful but not the way either of us thought!  I downloaded the latest zip file and compared those sources to what they had installed, and while they didn't match, all the diff's I checked said they only differed by source-code formatting conventions! Very strange. The other difference was their code was rooted in piwik, yours in matomo, so I did another search of their file system and found a piwik.zip, in with files from the Softaculous installer. Aha, I thought, Softaculous installs a forked version perhaps?  So I unpack that and compare and... it is identical to your matomo.zip, differing only that it is rooted in the current directory, yours in matomo/.

So I go into Softaculous via their WHM panel, and sure enough, there it is, it says piwik and installed at the time they said and assigned to the user where I found the files, but it says we have one minor number less than the current version, so I think, ok, let's see if it uses that zip I found, I click, and get an error that suggests permission problems, so I give up for the night.

I get email the next morning saying they can login!

So I can only guess what the fix really was, but I'd have never found it without your help!
